plugins/common/lambda-rules-registration.hpp

```diff
@@ -0,0 +1,131 @@
+#ifndef LAMBDARULESREGISTRATION_HPP
+#define LAMBDARULESREGISTRATION_HPP
+
+#include <map>
+#include <memory>
+#include <string>
+#include <tuple>
+
+#include "wayfire/object.hpp"
+#include "wayfire/nonstd/observer_ptr.h"
+#include "wayfire/rule/lambda_rule.hpp"
+
+namespace wf
+{
+
+struct lambda_rule_registration_t;
+
+using map_type = std::map<std::string, std::shared_ptr<lambda_rule_registration_t>>;
+
+/**
+ * @brief The lambda_rule_registration_t struct represents registration information for a single lambda rule.
+ *
+ * To make a registration, create one of these structures in a shared_ptr, fill in the appropriate values and
+ * register it on the lambda_rules_registrations_t singleton instance.
+ *
+ * At minimum, the rule string and if_lambda need to be set.
+ */
+struct lambda_rule_registration_t
+{
+    /**
+     * @brief rule This is the rule text.
+     *
+     * @note The registering plugin is supposed to set this value before registering.
+     */
+    std::string rule;
+
+    /**
+     * @brief if_lambda This is the lambda method to be executed if the specified condition holds.
+     *
+     * @note The registering plugin is supposed to set this value before registering.
+     */
+    wf::lambda_t if_lambda;
+
+    /**
+     * @brief else_lambda This is the lambda method to be executed if the specified condition does not hold.
+     *
+     * @note The registering plugin is supposed to set this value before registering.
+     * @note In most cases this should be left blank.
+     *
+     * @attention: Be very careful with this lambda because it will be executed on the signal for each view
+     *             that did NOT match the condition.
+     */
+    wf::lambda_t else_lambda;
+
+    /**
+     * @brief access_interface Access interface to be used when evaluating the rule.
+     *
+     * @note If this is left blank (nullptr), the standard view_access_interface_t instance will be used.
+     */
+    std::shared_ptr<wf::access_interface_t> access_interface;
+
+    /**
```

Comment:
If this is not supposed to be set by plugins, I can imagine we can declare it as private and add the other class as a friend?

Reply:
Done